brainiak/isfc.py

```diff
@@ -80,37 +86,50 @@ def isc(D, collapse_subj=True, return_p=False, num_perm=1000,
     n_vox = D.shape[0]
     n_subj = D.shape[2]
 
-    if return_p:
-        n_perm = num_perm
-    else:
-        n_perm = 0
-
-    ISC = np.zeros((n_vox, n_subj, n_perm + 1))
+    n_perm = num_perm*int(return_p)
+    max_null, min_null = gen_null_array(n_subj, n_perm,
+                                        collapse_subj, float_type)
+    ISC = np.zeros((n_vox, n_subj), dtype=float_type)

```

Comment:
Both here and in ISFC below, it is not obvious to me why you need separate code for the real and permuted computations - in the old version of the code, I just had a single loop that computed the real value on the first iteration, and then all the permutations.